doc: Software version proposal (#26)
atomone-hub/govgen-proposals: 002_SOFTWARE_VERSIONING.md

> It is the proposal submitter's understanding and hope that the GovGen
> community will be able to use the results of this and other Sentiment
> Proposals to develop in the future a binding decision-making/governance
> process to be determined (potentially including an upgrade to GovGen's
> software) with a goal of shaping the potential AtomOne chain. At this early
> stage, this and other Sentiment Proposals are anticipated to be introduced
> with an intention to better understand the GovGen community’s general
> preferences and views, and also to measure the ratio of active participants
> to help determine the quorum for future proposals.

## Context

AtomOne intends to be a minimalist chain to limit the attack surface.
Therefore, the choice of software versioning is critical to ensure good
reliability, along with a limited set of dependencies and modules.

## Decision

### Cosmos-SDK

The more important part of the dependency is the Cosmos-SDK, so it is the first
version we need to define.

While v0.45 is a battle-tested version and has proven to be viable, an [audit
driven by OAK Security][audit] in early 2024 has reported some serious threats
that have been fixed in v0.47. 

> On the other hand, v0.50 brings some interesting features in term of

Reviewer: ```suggestion
On the other hand, while v0.50 brings some interesting features in term of
```

> modularity, but it's not used by the CosmosHub yet (at the time of writing,

Reviewer: ```suggestion
modularity, it is not used by the CosmosHub yet (at the time of writing,
```

> Gaia v19 upgrade proposal is still in voting period).

Reviewer: Is this the only argument we want to bring for discarding 0.50.x? Might be challenged if put it this way. I think we need to put emphasis on the fact that newer versions have less guarantees in terms of reliability. Unless this was already discussed or decided to not mention, imho it should be.

Author: This is a good point, let's improve that part.

Author: @giunatale see aaa72667fb714518227a21776461fb6044e48342

In conclusion, we suggest that AtomOne depends on Cosmos-SDK v0.47.

### Tendermint/CometBFT

In order to ensure a good compatibility between the application and the
consensus engine, we suggest using the same version of the consensus engine
than the one used by Cosmos-SDK 0.47, which is CometBFT v0.37.

### Gaia

With the defined version of the Cosmos-SDK, this limits the choice of Gaia
codebase to fork to 4 versions:

- v15 fixes all the issues reported by the OAK audit
- v16 adds some IBC/ICS functionalities
- v17 upgrades ICS to 2.0
- v18 adds the wasmd and feemarket module

> Obviously, v18 is discarded because AtomOne won't have any VM functionality, as

Reviewer: ```suggestion
AtomOne will not have any VM functionality as
```

> specified in the draft of the [Constitution].

Reviewer: ```suggestion
specified in the draft of the [Constitution], thus v18 gets automatically discarded.
```

> Then, considering that the first version of AtomOne won't have IBC or ICS

Reviewer: ```suggestion
Considering that the first version of AtomOne will not have IBC or ICS
```

> enabled, there's no advantage in choosing to fork v16 or v17.

Reviewer: ```suggestion
enabled, there is no advantage in choosing to fork v16 or v17.
```

> This left out v15, which is the version we propose to fork for AtomOne.

Reviewer: ```suggestion
This leaves out only v15, which is the version we propose to fork for AtomOne.
```
